bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQDishWasherHandler.java

```diff
@@ -0,0 +1,168 @@
+        super(thing, stateDescriptionProvider, itemChannelLinkRegistry);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        lgThinqDishWasherApiClientService = LGThinQApiClientServiceFactory.newDishWasherApiClientService(lgPlatformType,
+                httpClientFactory);
+        channelGroupDashboardUID = new ChannelGroupUID(getThing().getUID(), CHANNEL_DASHBOARD_GRP_ID);
+        courseChannelUID = new ChannelUID(channelGroupDashboardUID, CHANNEL_WMD_COURSE_ID);
+        stateChannelUID = new ChannelUID(channelGroupDashboardUID, CHANNEL_WMD_STATE_ID);
+        processStateChannelUID = new ChannelUID(channelGroupDashboardUID, CHANNEL_WMD_PROCESS_STATE_ID);
+        remainTimeChannelUID = new ChannelUID(channelGroupDashboardUID, CHANNEL_WMD_REMAIN_TIME_ID);
+        doorLockChannelUID = new ChannelUID(channelGroupDashboardUID, CHANNEL_WMD_DOOR_LOCK_ID);
+    }
+
+    private void loadOptionsCourse(DishWasherCapability cap, ChannelUID courseChannel) {
+        List<StateOption> optionsCourses = new ArrayList<>();
+        cap.getCourses().forEach((k, v) -> optionsCourses.add(new StateOption(k, emptyIfNull(v.getCourseName()))));
+        stateDescriptionProvider.setStateOptions(courseChannel, optionsCourses);
+    }
+
+    @Override
+    public void updateChannelDynStateDescription() throws LGThinqApiException {
+        DishWasherCapability dwCap = getCapabilities();
+
+        List<StateOption> options = new ArrayList<>();
+        dwCap.getStateFeat().getValuesMapping()
+                .forEach((k, v) -> options.add(new StateOption(k, keyIfValueNotFound(CAP_DW_STATE, v))));
+        stateDescriptionProvider.setStateOptions(stateChannelUID, options);
+
+        loadOptionsCourse(dwCap, courseChannelUID);
+
+        List<StateOption> optionsDoor = new ArrayList<>();
+        dwCap.getDoorStateFeat().getValuesMapping()
+                .forEach((k, v) -> optionsDoor.add(new StateOption(k, keyIfValueNotFound(CAP_DW_DOOR_STATE, v))));
+        stateDescriptionProvider.setStateOptions(doorLockChannelUID, optionsDoor);
+
+        List<StateOption> optionsPre = new ArrayList<>();
+        dwCap.getProcessState().getValuesMapping()
+                .forEach((k, v) -> optionsPre.add(new StateOption(k, keyIfValueNotFound(CAP_DW_PROCESS_STATE, v))));
+        stateDescriptionProvider.setStateOptions(processStateChannelUID, optionsPre);
+    }
+
+    @Override
+    public LGThinQApiClientService<DishWasherCapability, DishWasherSnapshot> getLgThinQAPIClientService() {
+        return lgThinqDishWasherApiClientService;
+    }
+
+    @Override
+    protected Logger getLogger() {
+        return logger;
+    }
+
+    @Override
+    protected void updateDeviceChannels(DishWasherSnapshot shot) {
+        updateState("dashboard#" + CHANNEL_AC_POWER_ID,
+                (DevicePowerState.DV_POWER_ON.equals(shot.getPowerStatus()) ? OnOffType.ON : OnOffType.OFF));
```

Comment:
Simplify / make it better readable
```suggestion
                OnOffType.from(DevicePowerState.DV_POWER_ON.equals(shot.getPowerStatus())));
```